File: apps/mobile/androidApp/src/main/kotlin/co/heva/piro/android/main/MainTab.kt
package co.heva.piro.android.main

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.CalendarMonth
import androidx.compose.material.icons.outlined.NotificationsActive
import androidx.compose.material.icons.outlined.Settings
import androidx.compose.material.icons.outlined.Shield
import androidx.compose.ui.graphics.vector.ImageVector

/** The bottom-navigation destinations, adapted from the mockup to the Piro on-call domain. */
enum class MainTab(val label: String, val icon: ImageVector) {
    OnCall("On-call", Icons.Outlined.Shield),
    Alerts("Alerts", Icons.Outlined.NotificationsActive),
    Schedule("Schedule", Icons.Outlined.CalendarMonth),
    Settings("Settings", Icons.Outlined.Settings),
}

File: apps/mobile/iosApp/Piro/Main/MainTab.swift
import Foundation

/// The bottom-tab destinations, mirroring the Android `MainTab`. On-call and Alerts are real; Schedule is
/// a placeholder; Profile hosts the user's account, avatar, notification preferences and sign-out.
enum MainTab: Hashable, CaseIterable {
    case onCall, alerts, schedule, profile

    var title: String {
        switch self {
        case .onCall: return "On-call"
        case .alerts: return "Alerts"
        case .schedule: return "Schedule"
        case .profile: return "Profile"
        }
    }

    /// SF Symbol for the tab. Chosen to match the Android outlined icon set.
    var systemImage: String {
        switch self {
        case .onCall: return "shield"
        case .alerts: return "bell.badge"
        case .schedule: return "calendar"
        case .profile: return "person.crop.circle"
        }
    }
}
